api/v1alpha4/virtualmachinegroup_publishrequest_types.go

```diff
@@ -0,0 +1,196 @@
+// © Broadcom. All Rights Reserved.
+// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha4
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// VirtualMachineGroupPublishRequestConditionComplete is the Type for a
+	// VirtualMachineGroupPublishRequest resource's status condition.
+	//
+	// The condition's status is set to true only when all other conditions
+	// present on the resource have a truthy status.
+	VirtualMachineGroupPublishRequestConditionComplete = "Complete"
+)
+
+// VirtualMachineGroupPublishRequestSpec defines the desired state of a
+// VirtualMachineGroupPublishRequest.
+//
+// All the fields in this spec are optional. This is especially useful when a
+// DevOps persona wants to publish a VM Group without doing anything more than
+// applying a VirtualMachineGroupPublishRequest resource that has the same name
+// as said VMs in the same namespace as said VMs in VM Group.
```

Comment:
Should this be "as said **VM Group** in the same namespace"?

Reply:
I omitted that on purpose. It's a namespaced CRD. Seems obvious.